src/pages/index.js

```diff
@@ -1,205 +1,265 @@
 
-// Use Escape to close Modals
-function handleEscClose(e){
-    if (e.key == 'Escape') {
-        const modal = document.querySelector(".modal_opened");
-        closeModal(modal);
-    }
-}
+// Edit Profile Popup
+const profileEditPopup = new PopupWithForm(
+  "#profile-edit-modal",
+  handleProfileEditSubmit
+);
+profileEditPopup.setEventListeners();
 
 // Edit Button Modal
-function handleProfileEditSubmit(e) {
-    e.preventDefault();
-    profileName.textContent = profileNameInput.value;
-    profileDescription.textContent = profileDescriptionInput.value;
-    closeModal(profileEditModal);
-}
+profileEditBtn.addEventListener("click", () => {
+  const currentUserInfo = profileInfo.getUserInformation();
+  profileNameInput.value = currentUserInfo.name;
+  profileDescriptionInput.value = currentUserInfo.job;
+  profileEditPopup.open();
+});
 
-// Add card Modal
-function handleAddCardFormSubmit(e) {
-    e.preventDefault();
-    const name = cardTitleInput.value;
-    const link = cardUrlInput.value;
-    renderCard({name, link}, cardListEl);
-    // added the reset code below to reset the image name and url
-    e.target.reset();
-    cardFormValidator.disableSubmitButton();
-    closeModal(addCardModal);
-}
+// Add Card Popup
+const addCardPopup = new PopupWithForm(
+  "#add-card-modal",
+  handleAddCardFormSubmit
+);
+addCardPopup.setEventListeners();
+addNewCardButton.addEventListener("click", () => addCardPopup.open());
+
+// Image Preview Popup
+const previewPopup = new PopupWithImage("#preview-modal");
+previewPopup.setEventListeners();
+
+// Delete Confirmation Popup
+const deletePopup = new PopupWithConfirmation(
+  "#delete-picture-modal",
+  handleDeleteCard
+);
+deletePopup.setEventListeners();
+
+// Change Avatar Popup
+const changeProfilePopup = new PopupWithForm(
+  "#avatar-modal",
+  handleAvatarEditSubmit
+);
+changeProfilePopup.setEventListeners();
+editAvatar.addEventListener("click", () => changeProfilePopup.open());
 
 /**************************************************************************
- *                            EVENT LISTENERS                             *
+ *                               USER INFO                                *
  **************************************************************************/
 
-// Edit Button
-profileEditBtn.addEventListener("click", () => {
-    profileNameInput.value = profileName.textContent;
-    profileDescriptionInput.value = profileDescription.textContent;
-    openModal(profileEditModal)
+const profileInfo = new UserInfo({
+  nameSelector: ".profile__name",
+  jobSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
+api
+  .getUserInfo()
+  .then((userData) => {
+    profileInfo.setUserInformation({
+      name: userData.name,
+      job: userData.about,
+      avatar: userData.avatar,
+    });
+  })
+  .catch((err) => console.error(err));
+
+function handleProfileEditSubmit(formValues) {
+  profileEditPopup.setLoadingState(true);
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```